### ISSUE TITLE
Estimated dirichlet parameter out of bounded

### ISSUE THREAD

**Issue by FBoyang** [is0383kk/Dirichlet-VAE#1]
Hi, thanks for this great work. One thing I notice is that sometimes the autoencoder gives slightly negative estimated Dirichlet parameter. Do you have any insight into why that could happen? 

Thanks!

**Comment by is0383kk** [comment]
Thank you for your comment!

To be honest, I don't know the cause of it ... .
I have updated dir_vae.py to address possible misunderstandings.
The updated part is the forward function on line 144
Added two outputs to the return value of the forward function (gauss_z, dir_z).

In this Dir-VAE, the variable that follows a normal distribution using the Laplace approximation (gauss_z in the forward function) is used as input to the softmax function, which outputs a variable（dir_z in forward function） that follows a Dirichlet distribution.
In other words, I think that the Gaussian variable after the Laplace approximation (gauss_z in the forward function), which is the input to the softmax function, can potentially take on negative values.
Then, the variables that follow a Dirichlet distribution (dir_z in forward function), which is the output of the softmax function, should have a positive value with a sum of 1.

Is it possible that the Dirichlet parameter you mentioned refers to gauss_z?

**Comment by FBoyang** [comment]
Thank you very much for your reply. I think you are right! Thanks for the clarification
